torch/_dynamo/utils.py

```diff
@@ -299,6 +299,159 @@ def get_runtime_metrics_context() -> RuntimeMetricsContext:
     return _RUNTIME_METRICS_CONTEXT
 
 
+class CompileEventLogLevel(enum.Enum):
+    """
+    Enum that loosely corresponds with a "log level" of a given event.
+
+    CHROMIUM_EVENT: Logs only to tlparse.
+    COMPILE_EVENT: Logs to tlparse + PT2 Compile Events
+    COMPILATION_METRIC: Logs to tlparse, PT2 Compile Events, and dynamo_compile
+    """
+
+    CHROMIUM = 1
+    PT2_COMPILE = 2
+    COMPILATION_METRIC = 3
+
+
+class CompileEventLogger:
+    """
+    Helper class for representing adding metadata(i.e. columns) to various compile events.
+    Use CompileEventLogger.add_data() to add metadata to:
+    - Chromium events
+    - PT2 Compile Events
+    - CompilationMetrics
+
+    This should be used in conjunction with dynamo_timed(), which creates the various
+    events and contexts to add data to.
+    """
+
+    @staticmethod
+    def log_instant_event(
```

Comment:
One thing I've noticed is that out of the logging call sites you've updated:
- Only log_instant_event uses a CompileEventLogLevel param. In all other places, I think you're using one of the helper functions that are syntactic sugar.
- log_instant_event always gets a CompileEventLogLevel.CHROMIUM level.

I don't know why that strikes me as weird. If the syntactic sugar API is primarily what we're supposed to use for logging, maybe this should look the same? Does log_instant_event only make sense for CompileEventLogLevel.CHROMIUM?

Reply:
You can log instant events to PT2 Compile, though it's a rarely used feature. It logs as a separate row in the table. I think having it as the default option is reasonable, maybe I can add an extra function `log_instant_event_pt2_compile` or something so we don't need to pass log level directly.